sdk/common/src/jmh/java/io/opentelemetry/sdk/internal/StacktraceRenderBenchmark.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.internal;
+
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+/**
+ * This benchmark compares the performance of {@link StackTraceRenderer}, the custom length limit
+ * aware exception render, to the built-in JDK stacktrace renderer {@link
+ * Throwable#printStackTrace(PrintStream)}.
+ */
+@BenchmarkMode({Mode.AverageTime})
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Warmup(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+@Fork(1)
+@SuppressWarnings("StaticAssignmentOfThrowable")
+public class StacktraceRenderBenchmark {
+
+  private static final Exception simple = new Exception("error");
+  private static final Exception complex =
+      new Exception("error", new Exception("cause1", new Exception("cause2")));
+
+  static {
+    complex.addSuppressed(new Exception("suppressed1"));
+    complex.addSuppressed(new Exception("suppressed2", new Exception("cause")));
+  }
+
+  @State(Scope.Benchmark)
+  public static class BenchmarkState {
+
+    @Param Renderer renderer;
+    @Param ExceptionParam exceptionParam;
+
+    @Param({"10", "1000", "100000"})
+    int lengthLimit;
+  }
+
+  @SuppressWarnings("ImmutableEnumChecker")
+  public enum Renderer {
+    JDK(
+        (throwable, limit) -> {
+          StringWriter stringWriter = new StringWriter();
+          try (PrintWriter printWriter = new PrintWriter(stringWriter)) {
```

Comment:
Not sure if the overhead of opening a `PrintWriter` is negligible here, but you may want to move initialization and cleanup into two `@Setup` and `@TearDown` methods

Reply:
I'm going for an apples to apples comparison with the current implementation, which is defined [here](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/internal/AttributeUtil.java#L121-L128). Currently, we initialize a StringWriter / PrintWriter each and every time we add an exception to a span! 😬